### ISSUE TITLE
Fix music playing

### ISSUE THREAD

**Issue by Croydon** [inexor-game/code#88]
Steps to reproduce:
- set sound volume to 0
- set sound volume higher again
=> music is getting played again after 5 - 15 seconds

or:
- set sound volume to 0
- load a map
- set sound volume higher again
=> music isn't getting played at all

According to @Calinou the last behaviour is intended for faster loading speed, but this behaviour is nothing what players are expecting. 

In the second situation the music should getting loaded when the sound is getting turned on. Likely there is also a way to speed it up in the first scenario.

**Comment by Calinou** [comment]
The music could always be loaded to fix this, but then you're loading files for nothing when you disable music and keep it disabled.

**Comment by aschaeffer** [comment]
I think this wouldn't be much work:

* If sound volume changed
* and previous volume was 0
* and music was not loaded

-> load music

**Comment by lonnieg1214** [comment]
Hello guys, 

I am new to contributing to Git hub and I would like to try out this bug. Do we get assigned these bugs or does everyone just try to fix them? I also think that the volume should always be loaded since music is essentially apart of the game anyway. That way, once someone feels they want music they just turn it up and it is already there for them to hear and they get a consistent loading experience. Let me know what you think though but I would like to take this on.

**Comment by Fohlen** [comment]
Hello there @lonnieg1214 
You can pull a bug you favor, and start working on it.
Usually, during the process you upload your changes to a branch or directly file a pull-request and we will give you feedback on things to improve, how they fit best in the game, or what we like about it ofc!
@Croydon should we note this somewhere in the wiki, if we didn't do so already?

**Comment by a-teammate** [comment]
Hey there @lonnieg1214! nice to see you interested :)

Well the problem with this issue is that currently the sound was disabled in master :hankey:

We did this since we plan to switch the sound playback framework (see #337).  But since the last release we introduced a new way of handling dependencies (conan.io).

This particular dependency of the old way of playing sounds (SDL_mixer) was not covered by that package manager yet though, so we would have needed to create an own package.
We skipped that (since the packages for the new sound playback `openAL` and `ALMixer` are already there) and just disabled sound playback for now..

**So** i think there are 2 solutions:

1) either you use an older release (e.g. 0.8.0) to build inexor
but I don't know how easy that is, since our [wiki](https://github.com/inexorgame/code/wiki/Build) already contains the new commands. the wiki has a history though.
2) fix that bug in [sauerbraten](https://github.com/inexorgame/sauerbraten) (the game we forked) since the sound code didn't really change much. So we can just transfer it afterwards and know it works :)

I would choose 2)

**Comment by lonnieg1214** [comment]
Sorry about being new to all of this. But I have connected visual studio to my branch but what kind of solution would I open for this? I'm sure a win 32 console application would not work as I would like to test and replicate the issue and test once I have made changes. Thank you in advance.

**Comment by a-teammate** [comment]
I guess you are currently try to build Inexor, so I'd say you should try to follow https://github.com/inexorgame/code/wiki/Build and in case something is unclear please tell us in Riot (see [here where that is](https://github.com/inexorgame/code/wiki/Contact)).

Joining there would be of help anyway :)

**Comment by lonnieg1214** [comment]
I'll grab sauerbraten now I did read that it is easier to fix the issue in that.

Sent from my iPhone

On Jun 15, 2017, at 10:15 AM, Malte H <notifications@github.com<mailto:notifications@github.com>> wrote:


I guess you are currently try to build Inexor, so I'd say you should try to follow https://github.com/inexorgame/code/wiki/Build and in case something is unclear please tell us in Riot (see here where that is<https://github.com/inexorgame/code/wiki/Contact>).

Joining there would be of help anyway :)

**Comment by lonnieg1214** [comment]
Hey everybody, last question. How do I test once I have made changes? I have tried to simply run debug but it says that It could not find core textures  and will not compile.

**Comment by Croydon** [comment]
@lonnieg1214 Did you start Sauerbraten with the sauerbraten.bat (on Windows) or sauerbraten_unix file (on Linux and MacOS)?

**Comment by lonnieg1214** [comment]
No, I have not tried that Croydon. I will give that a go. I'm using windows so I'll use the bat file. Thanks for the quick reply too.

Sent from my iPhone

On Jun 17, 2017, at 9:15 PM, Croydon <notifications@github.com<mailto:notifications@github.com>> wrote:


@lonnieg1214<https://github.com/lonnieg1214> Did you start Sauerbraten with the sauerbraten.bat (on Windows) or sauerbraten_unix file (on Linux and MacOS)?

**Comment by Croydon** [comment]
@lonnieg1214 Your welcome. Starting it with the .bat file should solve the "Could not find core texture" problem.

However, you mentioned that you have also problems to compile. Where do you stuck there?

**Comment by lonnieg1214** [comment]
That actually was the only message that seemed to stop the compilation process. I'll give the bat file a try and see if something else comes up.

Sent from my iPhone

On Jun 17, 2017, at 9:20 PM, Croydon <notifications@github.com<mailto:notifications@github.com>> wrote:


@lonnieg1214<https://github.com/lonnieg1214> Your welcome. Starting it with the .bat file should solve the "Could not find core texture" problem.

However, you mentioned that you have also problems to compile. Where do you stuck there?

**Comment by Croydon** [comment]
The .bat file is only there for **starting** Sauerbraten, The compilation process is finished before you can start Sauerbraten.

What compiler do you use?

**Comment by lonnieg1214** [comment]
I'm only using visual studio. Sorry, but I'm not actually compiling it I'm just trying to debug it. Do I have to compile it to debug?

**Comment by Croydon** [comment]
@a-teammate Could you please help out here? I don't do debugging in Visual Studio so I'm not the best person the ask here.

General speaking, you need to compile it first in order to be able to debug it, but it is likely, that Visual Studio is first compiling it anyway, when you click on debugging.

@lonnieg1214 Where do you stuck right now?

**Comment by lonnieg1214** [comment]
Well I have been making changes and committing away and making steady changes to the sound.cpp file but for some reason it doesn't seem to affect it. I have confirmed that the file is changing but when I run the bat file and start the game it seems to still react the same way: changing the soundvol to 0 cuts the sound. When changing the soundvol above 0, you will have to then toggle the musicvol to get sound back. Again I am unsure if my changes are actually taking affect or not after running the bat file.

Sent from my iPhone

On Jun 25, 2017, at 1:53 PM, Croydon <notifications@github.com<mailto:notifications@github.com>> wrote:


@a-teammate<https://github.com/a-teammate> Could you please help out here? I don't do debugging in Visual Studio so I'm not the best person the ask here.

General speaking, you need to compile it first in order to be able to debug it, but it is likely, that Visual Studio is first compiling it anyway, when you click on debugging.

@lonnieg1214<https://github.com/lonnieg1214> Where do you stuck right now?

**Comment by a-teammate** [comment]
lonnieg1214: I'll be happy to help you to get sauerbraten to compile and work for you to solve this. :)
I think we should talk in https://riot.im/app/#/room/#inexor:matrix.org, just ping me there :)
